pilot/pkg/features/pilot.go

```diff
@@ -217,7 +217,7 @@ var (
 
 	EnableRemoteJwks = env.Register(
```

Comment:
I would convert this from a string to RemoteJwksMode here, so we don't need to do it so many times later.

Reply:
not following on `so we don't need to do it so many times later.`.
Will user not be able to set to arbitrary values if we do that? thought this is just a default value.

Reply:
This evaluates to a string on process init currently. Based on user input. My suggestion is to convert that string to a RemoteJwksMode.

`InsecureKubeConfigOptions` is an example of doing this (converting to a Set)

Reply:
Will do

Reply:
I think this will make `EnableRemoteJwks ` confusing if changed the type from bool to mode. Actually we should remove this env later. And only support `Mode`.

Reply:
This is for backwards compatibility so we don't need 2 vars